### PR TITLE
PROB-2053 Ecobee Integration Failing for Some Users

### PR DESCRIPTION
**Pull request by marstorp**
When Ecobee (connect) receives HttpResponseException with status 400 a "java.io.IOException: Stream closed" is thrown while accessing the error object. As a workaround a new string object is created using toString() on the response data.

Also changing refresh of authToken to only request refresh of the current used and to handle the new token when the method returns rather than have authToken call the method again, which can cause recursion.

### CODE REVIEW

File: smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy [hunk 1]
```diff
@@ -556,66 +546,87 @@ def poll() {
...
-						storeThermostatData(resp.data.thermostatList)
+				httpGet(pollParams) { resp ->
+					isThermostatPolled = true
+					if(resp.status == 200) {
+						if (thermostats || ecobeesensors) {
+							storeThermostatData(resp.data.thermostatList)
+						}
+						if (ecobeesensors) {
+							updateSensorData(resp.data.thermostatList.remoteSensors)
+						}
```
Comment (ron2911) [inline smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy:588]: should we add an `else` for other `resp.status` codes like we have for switches below?

Comment (marstorp) [reply smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy:588]: Other than 200 there shouldn't really be any other successful response statuses. If there wasn't any data updated during the poll, the health check coming in next PR will count this loss of data towards the 3 attempts before marking the device offline.
As log.warn only shows up in live logs they don't give that much information in prod.

File: smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy [hunk 2]
```diff
@@ -812,9 +825,10 @@ def toQueryString(Map m) {
 	return m.collect { k, v -> "${k}=${URLEncoder.encode(v.toString())}" }.sort().join("&")
 }
 
-private refreshAuthToken() {
+boolean refreshAuthToken() {
 	log.debug "refreshing auth token"
 	def notificationMessage = "is disconnected from SmartThings, because the access credential changed or was lost. Please go to the Ecobee (Connect) SmartApp and re-enter your account login credentials."
+	def status = false
```
Comment (ron2911) [inline smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy]: should we rename this to `isSuccess` or something more relevant?

File: smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy [hunk 3]
```diff
@@ -542,9 +527,14 @@ def purgeChildDevice(childDevice) {
 		log.info "No more thermostats to poll, unscheduling"
 		unschedule()
 		state.authToken = null
+		runIn(1, "killMe")
 	}
 }
 
+def killMe() {
+	app.delete()
```
Comment (ron2911) [inline smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy]: should we add `try {} catch {}` here just in case?

**Review by ron2911 — APPROVED** [review]
LGTM 👍

**Pull request merged**